lib/core/src/Cardano/Pool/Performance.hs

```diff
@@ -0,0 +1,278 @@
+      -- moving average according to the design spec.
+      --
+      -- To re-use existing code in "Cardano.Pool.Metrics" without drastic
+      -- changes take a Double here for now.
+      --
+      -- TODO: Implement exponential moving averaging according to the spec.
+    } deriving (Show, Eq)
+
+type Cost = Lovelace
+type Margin = Double
+newtype Rewards = Rewards Double
+
+withRewards :: Rewards -> (Double -> Double) -> Rewards
+withRewards (Rewards r) f = Rewards $ f r
+
+type Lovelace = Double
+
+-- The value
+-- 0.01 :: RelativeStakeOf "pool"
+-- would mean that a pool has a stake that is 1% of the total active stake in
+-- the epoch.
+newtype RelativeStakeOf (tag :: Symbol)
+    = RelativeStake { unRelativeStake :: Double }
+    deriving (Eq, Ord, Show)
+    deriving newtype (Num, Fractional)
+
+
+--------------------------------------------------------------------------------
+-- Formulas from spec
+--------------------------------------------------------------------------------
+
+-- A pool that never misses a block (that it is scheduled to produce) will yield
+-- maximal rewards.
+--
+-- When a_0 (influence of leader stake) is 0, this reduces down to just:
+--
+-- R * \sigma,
+--
+-- where R = totalRewards, \sigma = relativePoolStake
+maximalPoolRewards
+    :: Env
+    -> RelativeStakeOf "pool leader"
+    -> RelativeStakeOf "pool"
+    -> Rewards
+maximalPoolRewards env (RelativeStake leaderStake') (RelativeStake poolStake) =
+    let
+        a_0 = leaderStakeInfluence env
+        RelativeStake z_0 = saturatedPoolSize env
+        sigma' = min poolStake z_0
+        s' = min leaderStake' z_0
+    in Rewards $ (totalRewards env) / (1 + a_0)
+        * (sigma' + s'* a_0 * (sigma' - (s' * (z_0 - sigma') / z_0)) / z_0)
+
+-- | Actual pool rewards dependent on performance. Pools that miss blocks they
+-- are scheduled to produce receive less rewards.
+--
+-- Rewards are later distributed among its leaders and members. Use
+-- @poolMemberReward@ to calculate the rewards of a member delegating to the
+-- pool.
+--
+-- p. 38
+-- \hat{f}
+poolRewards
+    :: Env
+    -> RelativeStakeOf "pool leader"
+    -> RelativeStakeOf "pool"
+    -> Double
+    -> Rewards
+poolRewards env leaderStake' poolStake' recentAvgPerf =
+    withRewards (maximalPoolRewards env leaderStake' poolStake') $ \r ->
+        r * (recentAvgPerf ** gamma)
+ where
+   gamma = punishmentHarshness env
+
+-- Calculate the share of rewards of a member delegating to a pool.
+--
+-- From page 38 under
+-- 6.5.2 Reward Splitting inside a pool
+poolMemberReward
+    :: Rewards -- \hat{f}
+    -> Cost -- c
+    -> Margin  -- m
+    -> RelativeStakeOf "member" -- t
+    -> RelativeStakeOf "pool" -- \sigma
+    -> Double
+poolMemberReward
+    (Rewards rew)
+    cost'
+    margin'
+    (RelativeStake memberStake)
+    (RelativeStake totPoolStake) =
+    if rew > cost'
+    then (rew - cost') * (1 - margin') * (memberStake / totPoolStake)
+    else 0
```

Comment:
The calculation of pool and member rewards is way out of scope for this U/S. The calculation of `R` shouldn't force us to re-implement the whole reward mechanism. Rather, this is something we want to fetch from the node's.